### ISSUE TITLE
When I am training your model, the output feature map is often Inf. Why?

### ISSUE THREAD

**Issue by Elijah-Yi** [LoveSiameseCat/MoE-FFD#3]
thanks for your great work, When I am training your model, the output feature map is often Inf. Why?

**Comment by LoveSiameseCat** [comment]
Do you align with the version of the packages provided in the README? I tested this repo before uploading it online, so I believe it should work well. According to my experience,  If zero value appears in the denominator, it may cause an infinity value after backpropagation. U can check this issue by using “torch.isinf(tensor).any()”.

**Comment by Elijah-Yi** [comment]
I'm using the latest one. I found out that there was an Inf problem when executing self.proj in the Attention class that caused the Nan problem. Do you have any good ways to solve this problem?

**Comment by LoveSiameseCat** [comment]
The best way is to downgrade your PyTorch version. I suspect the NaN problem in the Attention class may be caused by a specific input, so another solution you can try is changing the random seed and checking if the frequency of the NaN issue changes. If it does, you'll need to identify the specific input.
